### PR TITLE
Add button component (twig)

### PR DESCRIPTION
**Pull request by TatianaFomina**
Adds configurable button "component" for twig templates. 

Resolves https://github.com/codex-team/codex.docs/issues/178

### CODE REVIEW

File: src/frontend/styles/vars.pcss [hunk 1]
```diff
@@ -7,6 +7,19 @@
   --color-bg-light: #f8f7fa;
```
Comment (neSpecc) [inline src/frontend/styles/vars.pcss:7]: в предыдущей строке `--color-button-danger` больше не нужен

Comment (TatianaFomina) [reply src/frontend/styles/vars.pcss:7]: meaning?

File: src/frontend/styles/components/button.pcss [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  font-weight: 500;
+  border: none;
+  cursor: pointer;
+  transition-property: background-color;
+  transition-duration: 0.1s;
+  border-radius: 8px;
+
+  @supports(-webkit-mask-box-image: url('')){
+    border-radius: 0;
+    -webkit-mask-box-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M0 10.3872C0 1.83334 1.83334 0 10.3872 0H13.6128C22.1667 0 24 1.83334 24 10.3872V13.6128C24 22.1667 22.1667 24 13.6128 24H10.3872C1.83334 24 0 22.1667 0 13.6128V10.3872Z' fill='black'/%3E%3C/svg%3E%0A") 48% 41% 37.9% 53.3%;;
```
Comment (neSpecc) [inline src/frontend/styles/components/button.pcss:18]: looks beatufull 💜

image

File: src/frontend/styles/components/button.pcss [hunk 3]
```diff
@@ -0,0 +1,86 @@
+.docs-button {
+  display: flex;
```
Comment (neSpecc) [inline src/frontend/styles/components/button.pcss:2]: maybe `inline-flex` would be better?

File: src/frontend/styles/components/button.pcss [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  transition-duration: 0.1s;
+  border-radius: 8px;
+
+  @supports(-webkit-mask-box-image: url('')){
+    border-radius: 0;
+    -webkit-mask-box-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M0 10.3872C0 1.83334 1.83334 0 10.3872 0H13.6128C22.1667 0 24 1.83334 24 10.3872V13.6128C24 22.1667 22.1667 24 13.6128 24H10.3872C1.83334 24 0 22.1667 0 13.6128V10.3872Z' fill='black'/%3E%3C/svg%3E%0A") 48% 41% 37.9% 53.3%;;
+  }
+
+  &__icon {
+    margin-right: 4px;
```
Comment (neSpecc) [inline src/frontend/styles/components/button.pcss:22]: Heroicons are fine, but their container has no internal paddings, so the left and right margins around the icon are not the same as at Figma

image

So lets increment these margins for a little. Or create a paddings inside the icon component.

File: src/frontend/styles/components/page.pcss [hunk 5]
```diff
@@ -40,11 +41,9 @@
     }
 
     &-button {
-      @apply --button;
-      @apply --button-primary;
-      padding: 5px 10px;
-      font-size: 13px;
-      margin-left: 10px;
+      margin-left: 20px;
+      display: inline-block;
```
Comment (neSpecc) [inline src/frontend/styles/components/page.pcss:45]: could be removed i guess

File: src/frontend/styles/vars.pcss (hunk 1 above)
Comment (neSpecc) [reply src/frontend/styles/vars.pcss:7]: I mean you have the `--color-button-warning` for this case. So we don't need the `--color-button-danger` anymore

**Comment by neSpecc** [comment]
Two similar icons. Lets leave the single one.

image

File: src/frontend/styles/vars.pcss [hunk 6]
```diff
@@ -3,10 +3,22 @@
...
   --color-link-active: #2071cc;
-  --color-button-danger: #ff5159;
   --color-bg-light: #f8f7fa;
   --color-page-active: #ff1767;
 
+  --color-button-primary: #3389FF;
+  --color-button-primary-hover: #2E7AE6;
+  --color-button-primary-active: #296DCC;
+
+  --color-button-secondary: #F3F6F8;
```
Comment (TatianaFomina) [inline src/frontend/styles/vars.pcss:13]: btw mockups did not contain grey button example, so, i picked random grey shade. @neSpecc can you please verify the color?

**Review by neSpecc — APPROVED** [review]
Sorry that I've touched your PR. I have been trying to resolve issues caused by my imperfections at the UI design process. So I've updated Figma icons and mockups itself, and then pushed new icons and dimesions to the branch to debug it.

**Review by robonetphy — APPROVED** [review]

**Pull request merged**